process/connections.go

```diff
@@ -24,16 +28,86 @@ func (m *CollectorConnections) GetTags(tagIndex int) []string {
+	if m.EncodedDNS != nil {
+		return GetDNS(m.EncodedDNS, addr.Ip)
+	} else if m.EncodedDnsLookups != nil {
+		first, offsets, err := GetDNSV2(m.EncodedDnsLookups, addr.Ip)
+		if err != nil {
+			return "", nil, err
+		}
+		firststring, err := getDNSNameFromListByOffset(m.EncodedDomainDatabase, int(first))
+		if err != nil {
+			return "", nil, err
+		}
+		var strings []string
+		if offsets != nil && (len(offsets) > 0) {
+			strings = make([]string, len(offsets))
+			for _, off := range offsets {
+				s, err := getDNSNameFromListByOffset(m.EncodedDomainDatabase, int(off))
+				if err != nil {
+					return "", nil, err
+				}
+				strings = append(strings, s)
+
+			}
+		}
+		return firststring, strings, nil
+	}
+	return "", nil, fmt.Errorf("No DNS encoded information")
 }
 
 // IterateDNS iterates over all of the DNS entries for the given addr, invoking the provided callback for each one
-func (m *CollectorConnections) IterateDNS(addr *Addr, cb func(i, total int, entry string) bool) error {
-	return IterateDNS(m.EncodedDNS, addr.Ip, cb)
+func (m *CollectorConnections) IterateDNS(addr *Addr, cb func(i, total int, entry string) bool) {
+	if m.EncodedDNS != nil {
+		IterateDNS(m.EncodedDNS, addr.Ip, cb)
+	} else if m.EncodedDnsLookups != nil {
+		if m.EncodedDomainDatabase != nil {
+			IterateDNSV2(m.EncodedDnsLookups, addr.Ip, func(i, total int, offset int32) bool {
+				s, err := getDNSNameFromListByOffset(m.EncodedDomainDatabase, int(offset))
+				if err == nil {
+					return cb(i, total, s)
+				}
+				return false
+			})
+		}
+	}
 }
 
 // GetDNSNames returns all the DNS entries
-func (m *CollectorConnections) GetDNSNames() []string {
-	return getDNSNames(m.EncodedDNS)
+func (m *CollectorConnections) GetDNSNames() ([]string, error) {
+	if m.EncodedDNS != nil {
+		return getDNSNames(m.EncodedDNS)
+	} else if m.EncodedDomainDatabase != nil {
+		return getDNSNameListV2(m.EncodedDomainDatabase), nil
+	}
+	return nil, fmt.Errorf("unknown dns names database")
+}
+
+// GetDNSV2 returns the DNS entries for the given addr.
+// The first argument returned is the first DNS entry followed by any additional name resolutions.  Most IPs will
+// have a single resolution so this dual format allows us to avoid allocations for the common case.  If there are
+// multiple name resolutions, there is no implied priority between the dual values
+// the return values are indexes into the dns database (GetDNSDatabase)
+
+func (m *CollectorConnections) GetDNSV2(addr *Addr) (int32, []int32, error) {
+	return GetDNSV2(m.EncodedDnsLookups, addr.Ip)
+}
+
+// IterateDNSV2 iterates over all of the DNS entries for the given addr, invoking the provided callback for each one
+func (m *CollectorConnections) IterateDNSV2(addr *Addr, cb func(i, total int, entry int32) bool) {
+	IterateDNSV2(m.EncodedDnsLookups, addr.Ip, cb)
+}
+
+// GetDNSNamesV2 returns all the DNS entries
+func (m *CollectorConnections) GetDNSNamesV2() []string {
```

Comment:
Same comment as above. This is no longer needed since `GetDNSNames` takes care of calling the right function.